logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/TypingIndicatorRepository.kt

```diff
@@ -30,30 +31,35 @@ import kotlinx.datetime.Clock
 import kotlinx.datetime.Instant
 
 internal interface TypingIndicatorRepository {
-    fun addTypingUserInConversation(conversationId: ConversationId, userId: UserId)
-    fun removeTypingUserInConversation(conversationId: ConversationId, userId: UserId)
+    suspend fun addTypingUserInConversation(conversationId: ConversationId, userId: UserId)
+    suspend fun removeTypingUserInConversation(conversationId: ConversationId, userId: UserId)
     suspend fun observeUsersTyping(conversationId: ConversationId): Flow<Set<ExpiringUserTyping>>
 }
 
 internal class TypingIndicatorRepositoryImpl(
-    private val userTypingCache: ConcurrentMutableMap<ConversationId, MutableSet<ExpiringUserTyping>>
+    private val userTypingCache: ConcurrentMutableMap<ConversationId, MutableSet<ExpiringUserTyping>>,
+    private val userPropertyRepository: UserPropertyRepository
 ) : TypingIndicatorRepository {
 
     private val userTypingDataSourceFlow: MutableSharedFlow<Unit> =
         MutableSharedFlow(extraBufferCapacity = BUFFER_SIZE, onBufferOverflow = BufferOverflow.DROP_OLDEST)
 
-    override fun addTypingUserInConversation(conversationId: ConversationId, userId: UserId) {
-        userTypingCache.safeComputeAndMutateSetValue(conversationId) { ExpiringUserTyping(userId, Clock.System.now()) }
-            .also {
-                userTypingDataSourceFlow.tryEmit(Unit)
-            }
+    override suspend fun addTypingUserInConversation(conversationId: ConversationId, userId: UserId) {
+        if (userPropertyRepository.getTypingIndicatorStatus()) {
+            userTypingCache.safeComputeAndMutateSetValue(conversationId) { ExpiringUserTyping(userId, Clock.System.now()) }
+                .also {
+                    userTypingDataSourceFlow.tryEmit(Unit)
+                }
+        }
     }
 
-    override fun removeTypingUserInConversation(conversationId: ConversationId, userId: UserId) {
-        userTypingCache.block { entry ->
-            entry[conversationId]?.apply { this.removeAll { it.userId == userId } }
-        }.also {
-            userTypingDataSourceFlow.tryEmit(Unit)
+    override suspend fun removeTypingUserInConversation(conversationId: ConversationId, userId: UserId) {
+        if (userPropertyRepository.getTypingIndicatorStatus()) {
```

Comment:
Should we leave this? If typing indicator gets disabled while we are typing, we should still allow disabling/removing the typing indicator in conversation right? Otherwise we could leave an undesired visual indicator hanging there forever.

Have you tested disabling it in a different client, while the current client is receiving and displaying some other users typing??